CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+   is broken on your local system::
+
+    pytest -vs pysatMadrigal
+
+6. Update/add documentation (in ``docs``).  Even if you don't think it's
+   relevant, check to see if any existing examples have changed.
+
+7. Add your name to the .zenodo.json file as an author
+
+8. Commit your changes and push your branch to GitHub::
+
+    git add .
+    git commit -m "Brief description of your changes"
+    git push origin name-of-your-bugfix-or-feature
+
+9. Submit a pull request through the GitHub website. Pull requests should be
+   made to the ``develop`` branch.
+
+Pull Request Guidelines
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need some code review or feedback while you're developing the code, just
+make a pull request. Pull requests should be made to the ``develop`` branch.
+
+For merging, you should:
+
+1. Include an example for use
+2. Add a note to ``CHANGELOG.md`` about the changes
+3. Ensure that all checks passed (current checks include Travis-CI
+   and Coveralls) [1]_
+
+.. [1] If you don't have all the necessary Python versions available locally or
+       have trouble building all the testing environments, you can rely on
+       Travis to run the tests for each change you add in the pull request.
+       Because testing here will delay tests by other developers, please ensure
+       that the code passes all tests on your local system first.
+
+Project Style Guidelines
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In general, pysat follows PEP8 and numpydoc guidelines.  Pytest runs the unit
+and integration tests, flake8 checks for style, and sphinx-build performs
+documentation tests.  However, there are certain additional style elements that
+have been settled on to ensure the project maintains a consistent coding style.
+These include:
+
+* Line breaks should occur before a binary operator (ignoring flake8 W503)
+* Combine long strings using `join`
+* Preferably break long lines on open parentheses rather than using `\`
+* Use no more than 80 characters per line
+* Avoid using Instrument class key attribute names as unrelated variable names:
+  `platform`, `name`, `tag`, and `inst_id`
+* The pysat logger is imported into each sub-module and provides status updates
+  at the info and warning levels (as appropriate)
+* Several dependent packages have common nicknames, including:
+  * `import datetime as dt`
+  * `import numpy as np`
+  * `import pandas as pds`
+  * `import xarray as xr`
+* All classes should have `__repr__` and `__str__` functions
+* Docstrings use `Note` instead of `Notes`
+* Try to avoid creating a try/except statement where except passes
+* Use setup and teardown in test classes
+* Use pytest parametrize in test classes when appropriate
+* Provide testing class methods with informative failure statements and
+  descriptive, one-line docstrings
+* Block and inline comments should use proper English grammar and punctuation
+  with the exception of single sentences in a block, which may then ommit the
+  final period
```

Comment:
Potential new addition
 - Avoid the use of `int` and `float` in favor of specifying a bit depth, such as `np.int64` or `np.float64` to ensure that all platforms have the same numerical precision.

Reply:
When you make this suggestion, could you also include an example since a lot of python doesn't involve casting?

Reply:
Don't's
```
var = int(iterable[0])
var2 = np.array(items, dtype=float)
for nptime in inst.index.values.astype(int):
    pass
```

Do's
```
var = np.int64(iterable[0])
var2 = np.array(items, dtype=np.float64)

for nptime in inst.index.values.astype(np.int64):
    pass

```

Reply:
ommit typo